src/useImageDimensions.ts

```diff
@@ -0,0 +1,36 @@
+import {useEffect, useState} from 'react'
+import {Image, ImageRequireSource} from 'react-native'
+
+export interface URISource {
+  uri: string
+}
+
+/**
+ * @param source either a remote URL or a local file resource.
+ * @returns original image width and height.
+ */
+function useImageDimensions(source: ImageRequireSource | URISource) {
+  const [state, setState] = useState<{
+    width?: number
+    height?: number
+    loading?: boolean
+    error?: any
+  }>({})
+  useEffect(() => {
+    if (typeof source === 'object' && typeof source.uri === 'string') {
```

Comment:
Instead of relying on `typeof`, I think that we should investigate if we could first try calling `Image.resolveAssetSource` and then if that doesn't return values we can call `Image.getSize`.

It's unclear from the documentation, but I have a hunch that `Image.resolveAssetSource` will return `undefined` or `null` for assets that requires an asynchronous operation.

If that is correct, we could set the initial value using `Image.resolveAssetSource` and then fall back to `Image.getSize` only when needed.

Reply:
Image.resolveAssetSource for remote image returns just strange object without width and height
Why shouldn't we use "typeof"? Is that bad?

Reply:
The problem with `typeof` is that it makes assumptions on how the internal object looks, I touched on this a bit more in another comment but by not doing that we can work with more platforms and potentially avoiding breakage on new platforms

Reply:
Without that check, there will be unexpected results. Especially for different platforms. I'll take responsibility to add the functionality if there will be any request.